app/models/order_cycle.rb

```diff
@@ -47,7 +47,7 @@ class OrderCycle < ActiveRecord::Base
     if user.has_spree_role?('admin')
       scoped
     else
-      where('coordinator_id IN (?)', user.enterprises)
+      where('coordinator_id IN (?)', user.enterprises.map(&:id))
```

Comment:
So your guess is that enterprises are loaded already and we don't need to start a special query here? Otherwise this would be more efficient:
```rb
where('coordinator_id IN (?)', user.enterprises.select("enterprises.id"))
```
It's difficult to judge in a scope though. We don't know who will be using the scope and what else they are doing.

Uh, I just realised that we are using the result in a sub-query here. So there is no additional query with `select`. It will just be inserted, right?
```sql
WHERE coordinator_id IN (SELECT enterprises.id FROM "enterprises" INNER JOIN ...)
```

In that case we potentially avoid querying the enterprises. In case of a big enterprise list, the sub query could also be much smaller than a long list of ids. At least, it would scale better.

Your thoughts on this?

Reply:
I thought about it, user.enterprises are lazy loaded already and map is better. but seeing @sauloperez comment above made me realize user.enterprises_ids would be better. is user.enterprises_ids valid?

Reply:
I checked, calling `*_ids` multiple times triggers a SQL query each time it is called while the association is not lazy loaded. But if the association is already lazy loaded, it uses the cache.

```
> user.recipe_ids ; nil
   (0.7ms)  SELECT "recipes".id FROM "recipes" WHERE "recipes"."user_id" = $1  [["user_id", 1]]
=> nil
> user.recipe_ids ; nil
   (0.7ms)  SELECT "recipes".id FROM "recipes" WHERE "recipes"."user_id" = $1  [["user_id", 1]]
=> nil
> user.recipes.to_a ; nil
  Recipe Load (0.9ms)  SELECT "recipes".* FROM "recipes" WHERE "recipes"."user_id" = $1  [["user_id", 1]]
=> nil
> user.recipe_ids 
=> [1]
```

Reply:
I'm quite uncomfortable that the scope is now making assumptions about whether or not the `enterprises` association is already lazy loaded. But yeah, happy about the way you do it here if the associations are already lazy-loaded anyway.

Reply:
maybe user.enterprise_ids would be better than map?

Reply:
Hmm... Yeah actually @luisramos0, sounds good to me.